### DISCUSSION TITLE
Taro v3.3.19 build H5 设置env development编译报错

### DISCUSSION THREAD

**Discussion by WilsonLau0755** [NervJS/taro#12075] [Q & A] [answered]
### 相关平台
H5

**浏览器版本: chrome**
**使用框架: Vue 3**

### 复现步骤
1、taro init初始化项目，选择Gitee中获取的默认模板进行项目初始化
2、在项目中taro build --type h5 --env development进行项目编译

### 期望结果
正常完成项目编译

### 实际结果
🙅 Failed to compile.

js/app.js from Terser
TypeError: t.replace is not a function
at Array.forEach ()

### 环境信息
```
  Taro CLI 3.3.19 environment info:
    System:
      OS: Windows 10
    Binaries:
      Node: 16.11.1 - C:\nvm\nodejs\node.EXE
      npm: 8.0.0 - C:\nvm\nodejs\npm.CMD

```

---
### 补充信息
仅编译H5时加上--env development会报错。

之前提过没解决关了的issue：https://github.com/NervJS/taro/issues/11467

**Comment by moyafaxi** [comment]
暂时先用 --watch 吧， env 就是 development 了,
现在命令行指定 env 有问题，暂时不要使用这种方式
@WilsonLau0755

**Comment by WilsonLau0755** [comment]
> 暂时先用 --watch 吧， env 就是 development 了, 现在命令行指定 env 有问题，暂时不要使用这种方式 @WilsonLau0755

主要是CI/CD用到，目前对于非prod环境只能手动构建然后上传云端。

**Comment by hqwlkj** [comment, marked as answer]
> 手

可以使用自定义环境变量来实现多环境的构建：

例如： 安装` cross-env `依赖


然后创建自己  CI/CD 的需要的环境，例如： `staging`， 然后在 config下创建 staging.js 文件
```js

// eslint-disable-next-line import/no-commonjs
module.exports = {
  env: {
    NODE_ENV: '"staging"', // 这里是你的环境变量
  },
  defineConstants: {
  },
  mini: {},
  h5: {
  }
}

```


再在 config/index.js 的 `module.exports` 中新增

```
  if (process.env.NODE_ENV === 'staging') { // staging 是你配置的 环境变量名称
    return merge({}, config, require('./staging'))
  }
```

然后在 `package.json` 的 `scripts` 新增 `"staging:h5": "cross-env NODE_ENV=staging yarn build:h5",`


修改你的 CI/CD 的大包命令为 ：`npm run staging:h5` 就好啦

**Reply by cnqi2023** [reply]
这个也不行的,vue3 build h5 就会报错

**Reply by shuichangdonga** [reply]
老哥解决了吗？我也是这个问题

**Reply by hqwlkj** [reply]
> 这个也不行的,vue3 build h5 就会报错

提示什么错误信息吗？

**Reply by shuichangdonga** [reply]
Error: js/app.js from Terser plugin
str.replace is not a function
TypeError: str.replace is not a function
    at Array.forEach (<anonymous>)

"build:dev": "cross-env NODE_ENV=development taro build --type h5",
"dev:dev": "npm run build:dev -- --watch"

npm run dev:dev可以正常运行，npm run build:dev就报错

**Reply by holycrypto** [reply]
react版本打包时候也报错

```
Error: chunk/712.js from Terser plugin
str.replace is not a function
TypeError: str.replace is not a function
    at Array.forEach (<anonymous>)
```

**Comment by SilentFlute** [comment]
可以看看这个: #14330
